### PR TITLE
Add force_rerun flag

### LINKED ISSUE #32: Allow for re-running prompts without using run_batch
## Description of the feature request

Hello, thank you for creating this library. As far as I can understand, due to caching it is not possible to run the same prompt/params multiple times and get a non-cached result

For the purpose of running prompts using parallel processing, or just generating results with the same input multiple times, it would be nice to be able to provide a flag which tells the client to re-run a prompt, and cache the result with a new key.

Perhaps have a re-run flag, which appends a unique string or param to the cache key, like a uuid or something? I'm happy to add this and PR it.

### PR DESCRIPTION
**Pull request by jmilldotdev**
This closes #32

When setting `force_rerun` to `True` on `client.run()`, a uuid will be generated and added to the cache key to create uniqueness and therefore allow the client to get a new generation from the chosen API.

### CODE REVIEW

File: manifest/manifest.py [hunk 1]
```diff
@@ -106,6 +107,7 @@ def run(
         prompt: Union[Prompt, str],
         input: Optional[Any] = None,
         gold_choices: Optional[List[str]] = None,
+        force_rerun: bool = False,
```
Comment (lorr1) [inline manifest/manifest.py:110]: So what does this do in comparison to `overwrite_cache`? Is there some reason you want to add an id?

Comment (jmilldotdev) [reply manifest/manifest.py:110]: Adding the `run_id` makes the cache key different, and therefore the results are written to the cache. This way you can run the same prompt+settings many times and cache all of the results.

Compare to `overwrite_cache`, where you can do many runs, but only the latest run is cached.

Comment (jmilldotdev) [reply manifest/manifest.py:110]: Perhaps the parameter should be named something different, like `cache_reruns`

**Comment by lorr1** [comment]
Addressed by adding run_id (in main).

**Pull request closed without merging**